src/Microsoft.Dnx.Runtime/DependencyManagement/CompatibilityChecker.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using Microsoft.Dnx.Runtime.DependencyManagement;
+using NuGet;
+
+namespace Microsoft.Dnx.Runtime
+{
+    public class CompatibilityChecker
+    {
+        private readonly LockFile _lockFile;
+        private readonly ILookup<string, LockFileLibrary> _lockFileLibraries;
+        private readonly Dictionary<FrameworkName, HashSet<string>> _frameworkRuntimeAssemblies;
+
+        public CompatibilityChecker(LockFile lockFile)
+            : this(new OptimizedLockFile(lockFile))
```

Comment:
Why do we need both? I don't trust that this won't be misused at runtime reducing the performance.